compiler/sigmatch.nim

```diff
@@ -730,7 +730,7 @@ proc matchUserTypeClass*(m: var TCandidate; ff, a: PType): PType =
       if oldWriteHook != nil: oldWriteHook msg
       diagnostics.add msg
 
-  var checkedBody = c.semTryExpr(c, body.copyTree, flags)
+  var checkedBody = c.semTryExpr(c, body.copyTree, flags + {efInConcept})
```

Comment:
I don't understand why concept bodies have to be treated differently in regard to this particular issue?

Reply:
I was a bit surprised by this, but here's an example from the tests:
```nim
type VectorSpace[K] = concept x, y
  x + y is type(x)
  zero(type(x)) is type(x)
  -x is type(x)
  x - y is type(x)
  var k: K
  k * x is type(x)
```